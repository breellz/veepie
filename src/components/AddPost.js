import React, { useEffect, useState } from 'react';
import{ connect } from 'react-redux'
import {firebase} from '../firebase/firebase'
//import Postform from './Postform'
import { startAddPost } from '../actions/posts'
import RichEditor from '../components/RichEditor'
import { database } from 'firebase';



const AddPost = (props) => {
  const [name, setName] = useState('Anonymous');
  const [uid, setUid] = useState('')
  useEffect(()=>{
    firebase.auth().onAuthStateChanged((user)=>{
      if (user) {
        setName(user.displayName)
        setUid(user.uid)
        }
     })
   },[]) 

  const onSubmit = ({...rest, tags}) => {
    const editedTags = tags.split(',')
    const post= {...rest, tags: editedTags, author: name, uid}
    props.startAddPost(post);
    props.history.push('/dashboard')
  }
 return(
  <div className="content-container add-post">
  
    <RichEditor  onSubmit = {onSubmit}/>
    </div>
 )
}

const mapDispatchToProps = (dispatch) => ({
  startAddPost : (post)=>(dispatch(startAddPost(post)))
})
export default connect(undefined, mapDispatchToProps)(AddPost);
