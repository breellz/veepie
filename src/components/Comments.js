import React from 'react';

const Comments = () => (
  <div>
    Comments page content
  </div>
);

export default Comments;
