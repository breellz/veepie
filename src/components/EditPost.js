import React from 'react';
import { connect } from 'react-redux'
import PostForm from './Postform';
import {editPost, deletePost} from '../actions/posts'


const EditPost = (props) => {
  console.log(props)
 const onRemove=()=>{
   props.deletePost(props.post.id)
   props.history.push('/')
 }
 const onSubmit = (post)=> {
   console.log(post)
   props.editpost(props.post.id, post)
  props.history.push('/')
}
  return(
  <div>
    <PostForm post = {props.post}
    onSubmit ={(post)=>onSubmit(post)}/>
    <button onClick={onRemove}>Delete Post</button>
  </div>
 )
};

const mapStateToProps = (state, props) => ({
  post: state.posts.find((post) => post.id === props.match.params.id)
});

const mapDispatchToProps = (dispatch, props) => ({
  editpost: (id, post) => dispatch(editPost(id, post)),
  deletePost: (id) => dispatch(deletePost(id))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditPost);
