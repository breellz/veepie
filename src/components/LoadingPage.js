import React from 'react';

const LoadingPage = () => (
  <div className="loader">
    <div className="loader-body"></div>
  </div>
);

export default LoadingPage;
