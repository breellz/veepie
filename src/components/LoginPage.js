import React from 'react';
import { connect } from 'react-redux';
import { startGoogleLogin} from '../actions/auth'



export const LoginPage = ({startGoogleLogin}) => (
  <div className= "login">
    <div className="login-body">
      <h2>Veepie</h2>
        <p>Tell a story...</p>
      <button className="login-body__button"
      onClick = { startGoogleLogin
      }>Login with Google</button>
    </div>
  </div>
);

const mapDispatchToProps = (dispatch)=> ({
  startGoogleLogin : ()=>  dispatch(startGoogleLogin()),
})

export default connect(undefined, mapDispatchToProps)(LoginPage);
