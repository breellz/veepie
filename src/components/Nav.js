import React from 'react';
import { NavLink } from 'react-router-dom';


export const Nav = () => (
  <header className="nav">
  <div className="content-container nav-link">
  <NavLink className="nav-link__item" to="/dashboard" activeClassName= "is-active"><h3>Dashboard</h3></NavLink>
  <NavLink className="nav-link__item" to ="/create" activeClassName ="is-active"><h3>Add Post</h3></NavLink><br />
  <NavLink className="nav-link__item" to ="/profile" activeClassName ="is-active"><h3>Profile</h3></NavLink>
  </div>
  </header>
);


export default Nav;
