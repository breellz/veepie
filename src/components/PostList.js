import React from 'react';
import { connect } from 'react-redux';
import PostsSelector from '../selectors/PostsSelector'
import PostlistItem from './PostlistItem'


const PostList = (props) => (
  <div>
  {
    props.posts.length === 0 ? (<p>No posts to show</p>) :
    (props.posts.map((post)=> <PostlistItem key={post.id} {...post} />))
   }
</div>
)

const mapStateToProps = (state) =>{
 return{
  posts: PostsSelector(state.posts, state.filters)
 } 
}
export default connect(mapStateToProps)(PostList);