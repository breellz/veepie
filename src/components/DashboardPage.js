import React from 'react';
import PostList from './PostList'
import PostlistFilters from './PostlistFilter'; 




const DashboardPage = (props) => (
  <div>
    <PostlistFilters />
    <PostList />
  </div>
);

export default DashboardPage

