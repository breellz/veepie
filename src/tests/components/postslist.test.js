import React from 'react'
import {shallow} from 'enzyme'
import Posts from '../fixtures/posts'
import {PostList} from '../../components/PostList'

test('should render postslist with posts', ()=>{
    const renderer = shallow(<PostList posts = {Posts}/>)
    expect(renderer).toMatchSnapshot()
})

test('should render postlist with empty message', ()=>{
    const wrapper = shallow(<PostList posts = {[]} />)
    expect(wrapper).toMatchSnapshot()
})