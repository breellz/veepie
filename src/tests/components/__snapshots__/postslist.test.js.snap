// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render postlist with empty message 1`] = `
<div>
  <p>
    No posts to show
  </p>
</div>
`;

exports[`should render postslist with posts 1`] = `
<div>
  <PostlistItem
    body="hola moi"
    comments={Array []}
    createdAt={1}
    id="whe12"
    key="whe12"
    tags={Array []}
    title="first post"
  />
  <PostlistItem
    body="oi"
    comments={Array []}
    createdAt={-1}
    id="whse12"
    key="whse12"
    tags={Array []}
    title="second post"
  />
  <PostlistItem
    body=" moi"
    comments={Array []}
    createdAt={0}
    id="whes2"
    key="whes2"
    tags={Array []}
    title="third post"
  />
</div>
`;
